File: SRC/Service/Saras.eMarking/Saras.eMarking.Domain/ViewModels/Project/MarkScheme/MarkSchemeModel.cs
using Microsoft.AspNetCore.Http;
using Saras.eMarking.Domain.Extensions;
using Saras.eMarking.Domain.ViewModels.File;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Saras.eMarking.Domain.ViewModels.Project.MarkScheme
{
    public class MarkSchemeModel
    {
        public MarkSchemeModel()
        {
        }

        public long? ProjectMarkSchemeId { get; set; }
        [XssTextValidation]

        public string SchemeCode { get; set; }
        [XssTextValidation]
        [Required]
        public string SchemeName { get; set; }
        [Required]
        public decimal? Marks { get; set; }
        
        [MaxLength(2000)]
        public string SchemeDescription { get; set; }
        public long? ProjectQuestionId { get; set; }
        public long CountOfTaggedQuestions { get; set; }
        [Required]
        public List<BandModel> Bands { get; set; }
        public bool Selected { get; set; }
        public int MarkScheme { get; set; }
        public bool IsBandExist { get; set; }
        public bool IsQuestionTagged { get; set; }
        public MarkSchemeType MarkSchemeType { get; set; }      
        public List<FileModel> filedetails { get; set; }
    }


    public class Filedetails
    {
        public long Id { get; set; }
        public string FileName { get; set; }
        public IFormFile FileContent { get; set; }
    }

}

File: SRC/Manifest/Saras.eMarkingClient/src/app/model/project/mark-scheme/mark-scheme-model.ts
import { FileModel } from "../../file/file-model";

export interface MarkScheme {
  ProjectMarkSchemeId: number;
  ProjectName: string;
  SchemeCode: string;
  SchemeName: string;
  Marks: number;
  questionID: number;
  QuestionCode: string;
  Bands: Band[];
  Selected: boolean;
  SchemeDescription: string;
  MarkSchemeType: MarkSchemeType;
  MarkScheme: number;
  IsBandExist: boolean;
  IsQuestionTagged: boolean;
  filedetails: FileModel[];
}
export interface Band {
  BandChk: boolean;
  BandName: string;
  BandFrom: number;
  BandTo: number;
  BandDescription: string;
}
export interface Questions {
  QuestionId: number;
  QuestionLabel: string;
  MaxScore: string;
  SchemeName: string;
  IsTagged: boolean;
  TotalRows: number;
}
export interface QuestionTag {
  ProjectQuestionId: number;
  QuestionId: number;
  MarkSchemeId: number;
  MaxMark: string;
  IsTagged: boolean;
  QuestionCode: string;
  PanelOpenState: boolean;
  IsScoringComponentExist: boolean;
  ComponentName: string;
}

export interface QuestionText {
  QuestionText: string;
}

export enum MarkSchemeType {
  QuestionLevel = 1,
  ScoreComponentLevel = 2,
}

export interface Filedetails {
  Id?: number;
  FileName: string;
  FileContent?: File;
}
